docs/concepts/improve_group_by_operation.md

```diff
@@ -43,67 +43,56 @@ The pandas API most likely cannot efficiently handle the complexity of the aggre
+Both approaches shown above return the exact same result, but Approach 1 is inefficient and returns the warning message
 we showed at the top.
 
 What makes the first approach inefficient and the second approach efficient? It comes down to what the
 pandas API lets us express.
 
 ## Approach 1
 
-```python
-# From line 11
-
-return df.group_by("a").agg((nw.col("b") + nw.col("c")).sum().alias("d"))
-```
-
-To translate this to pandas, we would do:
+The literal pandas translation is:
 
 ```python
 df.groupby("a").apply(
     lambda df: pd.Series([(df["b"] + df["c"]).sum()], index=["d"]), include_groups=False
 )
 ```
 
+pandas experts immediately recognise the issue: it uses `apply`.
 Any time you use `apply` in pandas, that's a performance footgun - best to avoid it and use vectorised operations instead.
 Let's take a look at how "approach 2" gets translated to pandas to see the difference.
 
 ## Approach 2
 
-```python
-# Line 11 in Approach 2
-
-return df.with_columns(d=nw.col("b") + nw.col("c")).group_by("a").agg({"d": "sum"})
-```
-
-This gets roughly translated to:
+The literal pandas translation is:
 
 ```python
 df.assign(d=lambda df: df["b"] + df["c"]).groupby("a").agg({"d": "sum"})
 ```
 
-Because we're using pandas' own API, as opposed to `apply` and a custom `lambda` function, then this is going to be much more efficient.
+Because we're using pandas' own API, as opposed to `apply` and a custom `lambda` function,
+then this is going to be much more efficient! That's why it's preferred.
 
 ## Tips for Avoiding the `UserWarning`
 
-To ensure efficiency and avoid warnings similar to those seen in Approach 1, we recommend that you follow these practices:
-
-1. Decompose complex operations: break down complex transformations into simpler steps. In this case, keep the `.agg` method simple. Compute new columns first, then use these columns in aggregation or other operations.
-2. Avoid redundant computations: if an operation (like addition) is used multiple times, compute it once and store the result in a new column.
-3. Leverage built-in functions: use built-in functions provided by the DataFrame library. In this case, using the `with_columns()` method allows you to pre-compute before grouping and aggregation.
-
-By following these guidelines, you can are sure to avoid the aforementioned warning.
+1. Decompose complex operations: break down complex transformations into simpler steps.
+   In this case, keep the `.agg` method simple.
+   Compute new columns first, then use these columns in aggregation or other operations.
+2. Avoid redundant computations: if an operation (like addition) is used multiple times,
+   compute it once and store the result in a new column.
 
-**_Happy grouping!_** 🫡
+In a future version of Narwhals, we might be able to do some of this automatically for
+you, even for backends like pandas which don't natively do query optimisation.
```

Comment:
1. I find happy grouping salute quite nice 😂
2. Are we... making some big promises here 👀?